### ISSUE TITLE
Terraform v0.11.15 pulls 2.2.0 by default instead of v1.2

### ISSUE THREAD

**Issue by mikemucc** [hashicorp/terraform-provider-external#100]
Terraform v0.11.15 pulls 2.2.0 instead of v1.2

### Terraform Version
0.11.15 (and other versions that use the 1.2 version)

### Affected Resource(s)
Running `terraform init` with version 0.11.15 and the external provider

### Terraform Configuration Files
```hcl
provider "external" { }
```
While using terraform 0.11.15

### Debug Output
```
/home # terraform -version
Terraform v0.11.15
...
Initializing provider plugins...
- Checking for available provider plugins on https://releases.hashicorp.com...
- Downloading plugin for provider "external" (2.2.0)...
```

### Expected Behavior
Terraform 0.11.15 downloads version 1.2 instead of 2.2.0

### Actual Behavior
Terraform 0.11.15 downloads version 2.2.0

### Steps to Reproduce
1. `terraform init`

### Important Factoids
Build logs show push failed for some part of the build/release:
https://github.com/hashicorp/terraform-provider-external/runs/4694624609?check_suite_focus=true

**Comment by Fullplate** [comment]
Terraform 0.12 appears to suffer from this issue too:
```
Provider "external" v2.2.0 is not compatible with Terraform 0.12.31.

Provider version 1.1.0 is the latest compatible version. Select it with
the following constraint:

    version = "~> 1.1"
```

The fix suggested in the error works:
```
provider "external" {
  version = "~> 1.1"
}
```

**Comment by bendavies** [comment]
> Terraform v0.11.15 pulls 2.2.0 instead of v1.2.5 (because 1.2.5 push to S3 Failed in yesterday's release build)

there is no version 1.2.5.
the new version is 2.2.0.

the release is now on s3. (https://github.com/hashicorp/terraform-provider-external/actions/runs/1655563351)

What i'm unclear on is why terraform 0.11/0.12 has started pulling 2.x instead of 1.1.x

**Comment by mikemucc** [comment]
I stand corrected and have updated the issue to reflect that.

**Comment by bflad** [comment]
Hi folks 👋  Thank you for reporting this. It should be resolved now for Terraform CLI 0.11. The fix for 0.12 will take a little longer.

**Comment by mikemucc** [comment]
Can you comment on what the issue/fix was?

**Comment by bflad** [comment]
Terraform CLI 0.10 and 0.11 perform provider discovery via HTTP headers, which need to be set during the release process. Without those headers, they will default to assuming compatibility. Since new versions of this provider are not compatible with those legacy CLI versions, the headers must be setup to exclude these new versions from discovery. You can see these via (now fixed):

```console
$  curl -I https://releases.hashicorp.com/terraform-provider-external/2.2.0/terraform-provider-external_2.2.0_darwin_amd64.zip
...
x-terraform-protocol-version: 5
x-terraform-protocol-versions: 5.0
...
```

Terraform CLI 0.12 depends on this same information being present in the Terraform Registry during its discovery process, but takes a restrictive approach. So while the new provider version is actually compatible with that Terraform CLI version, it will report a compatibility issue. You can see this via (not fixed yet):

```console
$ curl --silent 'https://registry.terraform.io/v1/providers/-/external/versions' | jq '.versions[] | select(.version == "2.1.0").protocols'
[
  "5.0"
]
$ curl --silent 'https://registry.terraform.io/v1/providers/-/external/versions' | jq '.versions[] | select(.version == "2.2.0").protocols'
[]
```

**Comment by nmische** [comment]
Any ETA on the Terraform Registry update to address the issue with Terraform CLI 0.12? Thanks!

**Comment by danw1979** [comment]
The compatibility (non-)issue with TF 0.12 has really clobbered us today - any idea when this might be fixed in the registry ?

**Comment by et304383** [comment]
Another company chiming in on this.  We're blocked!

**Comment by sfc-gh-emonsef** [comment]
We are blocked as well!  one temp solution is to pin the external version 
```
provider "external" {
  version = "2.1.1"
}
```
But interested also to know whats the ETA.

**Comment by justinlarose** [comment]
Last thing I heard was a new release was going to cut if Terraform Registry was not updated for 2.2.0 by this morning: https://github.com/goreleaser/goreleaser/issues/2768#issuecomment-1006041455

@bflad any new updates?

**Comment by bflad** [comment]
Hi again folks 👋  Thank you for your patience. The fix for Terraform CLI 0.12 is now in place as well:

```console
$ terraform0.12.31 init

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "external" (hashicorp/external) 2.2.0...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

There may be some lingering caching that will still return the old response, but it will resolve itself in that case shortly.
